stingray/crosscorrelation.py

```diff
@@ -0,0 +1,190 @@
+        self.n = len(self.corr)
+
+        self.time_shift, self.time_lags, self.n = self.cal_timeshift(dt=self.dt)
+
+    def cal_timeshift(self, dt=1.0):
+        """
+        Creates Crosscorrelation Object.
+
+        Parameters
+        ----------
+        dt: float , optional, default 1.0
+            Time resolution of lightcurve, should be passed when object is populated with correlation data
+            and no information about light curve can be extracted. Used to calculate time_lags.
+
+        Returns
+        ----------
+        self.time_shift: float
+             Value of time lag that gives maximum value of correlation between two lightcurves. 
+
+        self.time_lags: numpy.ndarray
+             An array of time_lags calculated from correlation data 
+        """
+
+        if self.dt is None:
+            self.dt = dt
+
+        if self.corr is None:
+            raise StingrayError('Time Shift cannot be calculated without correlation data')
+
+        self.n = len(self.corr)
+        dur = int(self.n / 2)
+        # Correlation against all possible lags, positive as well as negative lags are stored
+        x_lags = np.linspace(-dur, dur, self.n)
+        self.time_lags = x_lags * self.dt
+        # time_shift is the time lag for max. correlation
+        self.time_shift = self.time_lags[np.argmax(self.corr)]
+
+        return self.time_shift, self.time_lags, self.n
+
+    def plot(self, labels=None, axis=None, title=None, marker='-', save=False, filename=None):
+        """
+        Plot the :class:`Crosscorrelation` as function using Matplotlib.
+
+        Plot the Crosscorrelation object on a graph ``self.time_lags`` on x-axis and
+        ``self.corr`` on y-axis
+
+        Parameters
+        ----------
+        labels : iterable, default None
+            A list of tuple with xlabel and ylabel as strings.
+        axis : list, tuple, string, default None
+            Parameter to set axis properties of Matplotlib figure. For example
+            it can be a list like ``[xmin, xmax, ymin, ymax]`` or any other
+            acceptable argument for `matplotlib.pyplot.axis()` function.
+        title : str, default None
+            The title of the plot.
+        marker : str, default '-'
+            Line style and color of the plot. Line styles and colors are
+            combined in a single format string, as in ``'bo'`` for blue
+            circles. See `matplotlib.pyplot.plot` for more options.
+        save : boolean, optional (default=False)
+            If True, save the figure with specified filename.
+        filename : str
+            File name of the image to save. Depends on the boolean ``save``.
+        """
+
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError:
+            raise ImportError("Matplotlib required for plot()")
+
+        fig = plt.figure()
+
+        fig = plt.plot(self.time_lags, self.corr, marker)
```

Comment:
I'm a bit fuzzy on the syntax here. Are you creating a `fig` object twice? 
I don't pretend to understand in depth how matplotlib syntax works, but I tend to prefer the object-oriented syntax. If we implement it like that, we could allow the user to put in a `matplotlib.Axes` object or alternatively generate it with `fig, ax = plt.subplots(1, 1)`. You'd have to subsequently replace `plt.plot` with `ax.plot` and `plt.xlabel` with `ax.set_xlabel` and so on, but it has the advantage that it gives the user more control where the figure actually ends up being. For example, I could imagine a user might want to plot the cross correlation into a subplot of a larger multi-plot figure, and this would allow them to make sure the figure ends up exactly where they want it to be.